Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp

```diff
@@ -1899,26 +1898,44 @@ void NetworkResourceLoader::setWorkerStart(MonotonicTime value)
 void NetworkResourceLoader::startWithServiceWorker()
 {
     LOADER_RELEASE_LOG("startWithServiceWorker:");
+
+    auto newRequest = ResourceRequest { originalRequest() };
+#if ENABLE(CONTENT_FILTERING_IN_NETWORKING_PROCESS)
+    if (!startContentFiltering(newRequest))
+        return;
+#endif
+
     ASSERT(!m_serviceWorkerFetchTask);
-    m_serviceWorkerFetchTask = m_connection->createFetchTask(*this, originalRequest());
+    m_serviceWorkerFetchTask = m_connection->createFetchTask(*this, newRequest);
     if (m_serviceWorkerFetchTask) {
         LOADER_RELEASE_LOG("startWithServiceWorker: Created a ServiceWorkerFetchTask (fetchIdentifier=%" PRIu64 ")", m_serviceWorkerFetchTask->fetchIdentifier().toUInt64());
         return;
     }
 
-    serviceWorkerDidNotHandle(nullptr);
+    if (requestIsForServiceWorkerOnly())
+        return;
+
+    startRequest(newRequest);
```

Comment:
Can we just do `serviceWorkerDidNotHandle(nullptr);` here? It seems `start()` in `serviceWorkerDidNotHandle` will filter the content anyways?

Reply:
Thanks for reviewing! I will address these issues in the next patch.

Reply:
> Can we just do `serviceWorkerDidNotHandle(nullptr);` here? It seems `start()` in `serviceWorkerDidNotHandle` will filter the content anyways?

That is a good point. A feature of the content filter is that it can modify the URL being loaded. If we call `serviceWorkerDidNotHandle(nullptr);` here, we will load the original URL, regardless of whether the content filter modified the URL.

Thanks for reviewing!